diff --git a/scripts/build/debug/300-gdb.sh b/scripts/build/debug/300-gdb.sh
index e23298c7..b4699217 100644
--- a/scripts/build/debug/300-gdb.sh
+++ b/scripts/build/debug/300-gdb.sh
@@ -50,25 +50,6 @@ do_debug_gdb_build()
             cross_extra_config+=("--enable-build-warnings=,-Wno-format-nonliteral,-Wno-format-security")
         fi
 
-        # Target libexpat resides in sysroot and does not have
-        # any dependencies, so just passing '-lexpat' to gcc is enough.
-        #
-        # By default gdb configure looks for expat in '$prefix/lib'
-        # directory. In our case '$prefix/lib' resolves to '/usr/lib'
-        # where libexpat for build platform lives, which is
-        # unacceptable for cross-compiling.
-        #
-        # To prevent this '--without-libexpat-prefix' flag must be passed.
-        # Thus configure falls back to '-lexpat', which is exactly what we want.
-        #
-        # NOTE: DO NOT USE --with-libexpat-prefix (until GDB configure is smarter)!!!
-        # It conflicts with a static build: GDB's configure script will find the shared
-        # version of expat and will attempt to link that, despite the -static flag.
-        # The link will fail, and configure will abort with "expat missing or unusable"
-        # message.
-        extra_config+=("--with-expat")
-        extra_config+=("--without-libexpat-prefix")
-
         # ct-ng always builds ncurses in cross mode as a static library.
         # Starting from the patchset 20200718 ncurses defines a special macro
         # NCURSES_STATIC for a static library. This is critical for mingw host
@@ -175,8 +156,12 @@ do_debug_gdb_build()
         # version of expat and will attempt to link that, despite the -static flag.
         # The link will fail, and configure will abort with "expat missing or unusable"
         # message.
+        # !PATCH!: The above comment in not true: --without-libexpat-prefix doesn't
+        # really solve the issue and GDB still defaults to /usr/lib
         native_extra_config+=("--with-expat")
-        native_extra_config+=("--without-libexpat-prefix")
+        native_extra_config+=("--with-libexpat-prefix=${CT_SYSROOT_DIR}/usr")
+        # !PATCH!: same applies for libgmp
+        native_extra_config+=("--with-libgmp-prefix=${CT_SYSROOT_DIR}")
 
         do_gdb_backend \
             buildtype=native \
